### DISCUSSION TITLE
how can i easily extend?

### DISCUSSION THREAD

**Discussion by birdmichael** [x-0o0/ChatUI#16] [Q&A] [answered]
I noticed that your code uses enumerations extensively, such as MediaType and MessageOption. However, as a library, how can one easily extend and add their own functionalities to them?

**Comment by x-0o0** [comment, marked as answer]
Hi, @birdmichael ! Thanks for letting me know your issue.

🤔 Unfortunately, current enumerations *cannot* be extended. 

In 1.0.0 beta version, the goal 🎯 is to support _**"the most basic and common"**_ message options or media types. Once the basic features are implemented, I plan to work on the extensibility of the types defined as enums.  For example, I'm considering changing `MessageOption` or `MediaType` to `struct` types.

(Since it is an open source project, you are also **_welcome to submit a pull request_** directly without waiting for an update. 👍)

💡 I will definitely keep you informed about any work related to this. 

🎯 By the way, ChatUI aims ...
- to release **beta** versions **_every week_**. 🚀
- to release version `1.0.0` with fully implemented "Chat in Channel" feature by _**June**_. 🚀

If you have any issues or more questions, please let me know freely! 🙂

**Comment by birdmichael** [comment]
그리고, 가능한 한 많이 나눠야 한다고 생각합니다.
예를 들어, giphy를 미리 만듭니다.
